### DISCUSSION TITLE
Automatically run login or reauth when required.

### DISCUSSION THREAD

**Discussion by leoleoasd** [kanidm/kanidm#1690] [Ideas]
When encouting `Session has expired` or `Privileges have expired`, we can automatically run `kanidm login` or `kanidm reauth`. Since both these need interactive confirm (by typing password or touching passkey), we won't automatically do anything bad.

May be we can fork a subprocess that run `login/reauth` on token expirations, wait for the process to exit and check again? If it failed again we exit (since user may Ctrl-C -ed) and if the token is active now we proceed the current action.

**Comment by yaleman** [comment]
This would be a good idea as far as I'm concerned, but the process flow sure needs some discussion... I wonder if it's possible to make this happen neatly within clap's internals

ie, 

1. if the user does "kanidm <command> <arguments>" and the token's expired
2. effectively run `kanidm login -d <username>`
3. start the original command again

**Comment by leoleoasd** [comment]
Maybe it is possible to generate a `LoginOpt` and call it's `exec` from other commands?
